ryven/main/Ryven.py

```diff
@@ -1,105 +1,553 @@
+             '(default: %(default)s)')
+
+    group.add_argument(
+        '-f', '--flow-theme',
+        choices=[
+            'toy', 'tron', 'ghost', 'blender', 'simple', 'ueli',
+            'pure dark', 'colorful dark',
+            'pure light', 'colorful light', 'industrial', 'fusion'],
+        dest='flow_theme',
+        help='set the theme of the flow view '
+             '(default: {pure dark|pure light}, depending on the window theme)')
+
+    group.add_argument(
+        '--performance',
+        choices=['pretty', 'fast'],
+        default='pretty',
+        dest='performance',
+        help='select performance mode '
+             '(default: %(default)s)')
+
+    # TODO: Python >= 3.9
+    # group.add_argument(
+    #     '--animations',
+    #     action=argparse.BooleanOptionalAction,
+    #     dest='animations',
+    #     help='en/disable animations '
+    #          '(default: %(default)s)')
+    exclusive_group = group.add_mutually_exclusive_group()
+    exclusive_group.add_argument(
+        '--no-animations',
+        action='store_false',
+        dest='animations',
+        help='do not use animations '
+             '(default: use animations)')
+    exclusive_group.add_argument(
+        '--animations',
+        action='store_true',
+        dest='animations',
+        help='use animations '
+             '(default: use animations)')
+
+    # TODO: Python >= 3.9
+    # group.add_argument(
+    #     '--info-messages',
+    #     action=argparse.BooleanOptionalAction,
+    #     dest='info_messages',
+    #     help='en/disable info messages '
+    #          '(default: %(default)s)')
+    exclusive_group = group.add_mutually_exclusive_group()
+    exclusive_group.add_argument(
+        '--info-messages',
+        action='store_true',
+        dest='info_messages',
+        help='show info messages '
+             '(default: do not show info messages')
+    exclusive_group.add_argument(
+        '--no-info-messages',
+        action='store_false',
+        dest='info_messages',
+        help='do not show info messages '
+             '(default: do not show info messages')
+
+    # Debug/Verbose
+
+    group = parser.add_argument_group('verbose')
+
+    group.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        dest='verbose',
+        help='display messages on the console')
+
+    # FIXME: Is there a way to get the version dynamically?
+    # Could also be used in `MainWindow.save_project()`
+    group.add_argument(
+        '-V', '--version',
+        action='version',
+        version=...) #'%(prog)s 3.1')
+
+    group.add_argument(
+        '-q', '--qt-api',
+        default='pyside2',
+        dest='qt_api',
+        help='the QT API to be used '
+             '(default: %(default)s)')
+
+    group.add_argument(
+        '-t', '--title',
+        default='Ryven',
+        dest='title',
+        help="changes the window's title "
+             '(default: %(default)s)')
```

Comment:
why are `version`, `qt_api` and `title` part of the `verbose` group, was this intended?

Reply:
Once upon a time … this was called `debugging` and there was no `--version` parameter. I thought, that specifying the Qt API falls under the category of debugging and as for the title, this admittedly is a borderline case. All this does not make sense, if the group is now called `verbose`.

Should it be renamed back to `debugging`?

On the other hand, `--verbose` and `--version` could go into the general group, `--title` into the `display` group, which leaves us with `--qt-api`, which somehow would also fit into the `display` group?

Reply:
> On the other hand, `--verbose` and `--version` could go into the general group, `--title` into the `display` group, which leaves us with `--qt-api`, which somehow would also fit into the `display` group?

I think this is good! For `--qt-api`: definitely fits `display`, no? I plan to provide seamless PySide6 support, so the user might also generally choose e.g. to use `pyside6` API instead, so I think both `display` and general work perfectly fine.

Reply:
a0dc50c7eaadd440de5f8af4b1420e7aaa82fc72